### ISSUE TITLE
Bug(sort-imports): Misleading `Missed spacing` errors

### ISSUE THREAD

**Issue by hugop95** [azat-io/eslint-plugin-perfectionist#322]
### Describe the bug

Not a breaking bug but can be a minor inconvenience: `Missed spacing` errors may be reported way more than what the sorted result would require. See code below

### Code example

Configuration:

```
{
  "newlinesBetween": "always",
  "groups": [
    "external", "unknown"
  ]
}
```

Code:
```ts
import a1 from "./a"
import a2 from "a"
import b1 from "./b"
import b2 from "b"
import c1 from "./c"
import c2 from "c"
import d1 from "./d"
import d2 from "d"
```

The sorted result is:
```ts
import a2 from "a"
import b2 from "b"
import c2 from "c"
import d2 from "d"

import a1 from "./a"
import b1 from "./b"
import c1 from "./c"
import d1 from "./d"
```

but the following `Missed spacing` are reported (order-errors are not shown here):
```
  3:1  error  Missed spacing between "a" and "./b" imports            perfectionist/sort-imports
  5:1  error  Missed spacing between "b" and "./c" imports            perfectionist/sort-imports
  7:1  error  Missed spacing between "c" and "./d" imports            perfectionist/sort-imports
```

### ESLint version

8.57.0

### ESLint Plugin Perfectionist version

3.8.0

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by azat-io** [comment]
I think that's normal behavior. Don't you think so?

**Comment by hugop95** [comment]
@azat-io It is misleading in my opinion because it tells the user to make fixes that in the end will be erased when the imports will eventually be sorted.

I think that each error reported (and as such fix) should bring the code closer to the expected state.

That's how I see it, but it's nitpicking, I understand if you see it differently!

@OlivierZal, what do you think?

**Comment by OlivierZal** [comment]
I understand both points of view.

My way of using eslint (preference for autofix) would make me vote for @hugop95's proposal. 

However I think it's good to warn the users about the errors he actually has, and not about what they _could_ be after a _potential_ fix. 

I say potential because maybe some developers don't enable autofix, so they could manually based on (all) the error messages. 

In the example given in the previous comment, maybe the fix wouldn't be the one proposed, but another one based on the missing space warnings:

```ts
// comment separator
import a1 from "./a"
import a2 from "a"

// comment separator
import b1 from "./b"
import b2 from "b"

// comment separator
import c1 from "./c"
import c2 from "c"

// comment separator
import d1 from "./d"
import d2 from "d"
```

Then displaying all the warning messages could have some interest, so the user could fix accordingly as wanted.

**Comment by OlivierZal** [comment]
I understand both points of view.

My way of using eslint (preference for autofix) would make me vote for @hugop95's proposal. 

However I think it's good to warn the users about the errors he actually has, and not about what these errors _could_ be after a _potential_ fix. 

I say potential because maybe some developers don't enable autofix, so they could fix manually based on (all) the error messages. 

In the example given in the previous comment, maybe the fix wouldn't be the one proposed, but another one based on the missing space warnings:

```ts
// comment separator
import a1 from "a"

import a2 from "./a"

// comment separator
import b1 from "b"

import b2 from "./b"

// comment separator
import c1 from "c"

import c2 from "./c"

// comment separator
import d1 from "d"

import d2 from "./d"
```

Then displaying all the warning messages could have some interest, so the user could fix accordingly as wanted. 

Again, it's not my way to proceed, so maybe my example is too fictive to be relevant.

**Comment by hugop95** [comment]
Closing for now due to different opinions regarding the issue 😁
